openbao.yaml

```diff
@@ -19,25 +19,23 @@ environment:
 pipeline:
   - uses: git-checkout
     with:
+      # checking out to a subdirectory prevents git status to report dirty and
+      # bao --version reporting the version as dirty
+      destination: 'src'
       repository: https://github.com/openbao/openbao
       tag: v${{package.version}}
       expected-commit: 96853bb4de27ab8ffd1b0c2898c691460d43edeb
 
   - runs: |
-      # echo ".cache" >>.git/info/exclude
-      # echo ".config" >>.git/info/exclude
-      # echo ".gitconfig" >>.git/info/exclude
-      # echo ".npm" >>.git/info/exclude
-      # echo ".yarn" >>.git/info/exclude
-      # echo "melange-out" >>.git/info/exclude
       export LD_FLAGS="-X github.com/openbao/openbao/version.fullVersion=${{package.version}}"
-      make ember-dist bin GOARCH=$(go env GOARCH)
+      make -C src ember-dist
```

Comment:
Just a note but we don't yet support the Web UI because it still contains HashiCorp branded assets. I'd suggest not building it until a release with the UI re-enabled is released.

Reply:
The problem is building the binary fails if the UI is not present:
```
http/assets.go:16:12: pattern web_ui/*: no matching files found
```

It's possible to workaround this by creating an empty file called `http/web_ui/index.html`, but that seems like a hacky solution and we'd need to re-add `make ember-dist` in the next release.

When is this planned to be updated so we can ship the UI? Perhaps we can consider holding off on merging this and building images until the UI is ready?

If not, then we can go ahead with an empty index.html file, but is it possible to let us know once we should start including the UI?

Reply:
Ahh, I see, this is because [`make bin`](https://github.com/openbao/openbao/blob/main/Makefile#L26-L28) does a `ui`-enabled build. We should fix that upstream, but my suggestion to [others doing distributions](https://github.com/openbao/openbao/issues/431#issuecomment-2252912056) was just to call `go build` directly as that's what the go-releaser tooling we use for releases does.

You'll have to plumb through the hashes and commit date though, but you'll want to use these over build time for reproducibility. I've done something similar for GitLab here: https://gitlab.com/gitlab-org/govern/secrets-management/openbao-internal/-/blob/main/Makefile?ref_type=heads#L21 

(Though, note that we're doing a patching based flow, so you won't need the two separate commits which complicates things.)

Reply:
@cipherboy Good point that we can just use `go build` directly and pass the version details instead of calling Makefile that in turn calls the script and passes the `ui` tag. I switched to this approach just now.